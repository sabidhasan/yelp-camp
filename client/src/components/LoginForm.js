import React from 'react'

const LoginForm = (props) => {
  return (
    <div className='login-background'>
      <div className='login-form'>
        <h1>Log In</h1>
        <a className='login-form__close' onClick={() => props.toggleLoginForm()}>X</a>

        <button>Google Login</button>
        <button>Facebook Login</button>

      </div>
    </div>
  )
}

export default LoginForm
