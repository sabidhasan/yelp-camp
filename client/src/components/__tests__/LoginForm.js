import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import LoginForm from '../LoginForm'

configure({ adapter: new Adapter() });

describe('<LoginForm />', () => {
  let component;
  let props;
  const signIn = jest.fn();

  beforeEach(() => {
    component = shallow(<LoginForm { ...props } />, {context: {signIn}})
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Google Sign In should be called when button pressed', () => {
    component.find('.LoginForm__google').simulate('click');
    expect(signIn).toHaveBeenCalled();
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
