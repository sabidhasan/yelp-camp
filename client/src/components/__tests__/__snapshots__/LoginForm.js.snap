// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> LoginForm component snapshot matches what is expected 1`] = `
<div
  className="LoginForm flex-center"
>
  <div
    className="LoginForm__content"
  >
    <h1
      className="LoginForm__title"
    >
      Sign Up or Sign In to YelpCamp
    </h1>
    <XButton
      className="LoginForm__close bold"
      onClick={[Function]}
    />
    <p
      className="LoginForm__text"
    >
      We support logging in through 
      <a
        href="https://www.google.com"
      >
        Google
      </a>
       or 
      <a
        href="https://www.facebook.com"
      >
        Facebook
      </a>
      .
      <br />
      This is used verification, spam prevention and for your name if you choose to post a public comment. No other data is given to us by either provider.
      <br />
      Your data will not be kept on YelpCamp computers.
    </p>
    <button
      className="LoginForm__login-button btn btn--small LoginForm__google"
      onClick={[Function]}
    >
      <i
        className="fab fa-google"
      />
      Google Login
    </button>
    <button
      className="LoginForm__login-button btn btn--small LoginForm__facebook"
      disabled={true}
      onClick={[Function]}
    >
      <i
        className="fab fa-facebook-square"
      />
      Facebook Login
    </button>
  </div>
</div>
`;
